infra/build/functions/target_experiment.py

```diff
@@ -81,6 +82,20 @@ def run_experiment(project_name, target_name, args, output_path,
   local_stacktrace_path = os.path.join(build.out, 'stacktrace/')
   fuzzer_args = ' '.join(args + [f'-artifact_prefix={local_artifact_path}'])
 
+  # Upload err.log.
+  if errlog_path:
+    compile_step_index = next((i for (i, d) in enumerate(steps)
```

Comment:
1. nit: `i, step` for clarity.
2. Check `'args'` is a key in `d` before calling `d['args']`.
3. nit: Do we need `next((` or is `next(` also valid? 

If these changes made the line too long, we can expand this into a `for`-loop for better readability:
```python
compile_step_index = -1
for i,step in enumerate(steps):
  if step.get('args') and len(step.get('args')) >2 and '&& compile' in step.get('args')[2]:
    compile_step_index = i
    break
```

Reply:
Do we need the index here `step.get('args')[2]`
Just check all args would probably be more resilient to future changes.

Reply:
OK good point